processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -91,6 +117,59 @@ func (tsp *tracesamplerprocessor) Shutdown() error {
 	return nil
 }
 
+// parseSpanSamplingPriority checks if the span has the "sampling.priority" tag to
+// decide if the span should be sampled or not. The usage of the tag follows the
+// OpenTracing semantic tags:
+// https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
+func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
+	attribMap := span.GetAttributes().GetAttributeMap()
+	if attribMap == nil {
+		return deferDecision
+	}
+
+	samplingPriorityAttrib := attribMap["sampling.priority"]
+	if samplingPriorityAttrib == nil {
+		return deferDecision
+	}
+
+	decideForDoubleFn := func(value float64) samplingPriority {
+		if value == 0.0 {
+			return doNotSampleSpan
+		} else if value > 0.0 {
+			return mustSampleSpan
+		}
+		return deferDecision
+	}
```

Comment:
Is this function worth the effort? It only saves 2 lines and adds cognitive overhead. I'm also not sure whether the Go compiler will inline the call.

Reply:
The compiler does inline this call, but, let me put the alternative and see what people prefer in general.